src/installer/tests/HostActivation.Tests/DotnetArgValidation.cs

```diff
@@ -68,6 +68,40 @@ public void MissingArgumentValue_Fails()
                 .And.HaveStdErrContaining($"Failed to parse supported options or their values:");
         }
 
+        [Fact]
+        public void File_ExistsNoDirectorySeparator_RoutesToSDK()
+        {
+            // Create a file named "build" in the current directory to simulate a file that happens to match the name of a command
+            string buildFile = Path.Combine(sharedTestState.BaseDirectory.Location, "build");
+            File.WriteAllText(buildFile, string.Empty);
+
+            // Test that "dotnet build" still routes to SDK, not to the file
+            TestContext.BuiltDotNet.Exec("build")
+                .WorkingDirectory(sharedTestState.BaseDirectory.Location)
+                .EnableTracingAndCaptureOutputs()
+                .Execute()
+                .Should().Fail()
+                .And.HaveStdErrContaining("The command could not be loaded, possibly because:") // This is a generic error for when we can't tell what exactly the user intended to do
+                .And.HaveStdErrContaining("Resolving SDKs");
+        }
+
+        [Fact]
+        public void RelativePathToNonManagedFile_ShowsSpecificError()
+        {
+            // Test relative path with directory separator
+            Directory.CreateDirectory(Path.Combine(sharedTestState.BaseDirectory.Location, "subdir"));
+            string testFile = Path.Combine(sharedTestState.BaseDirectory.Location, "subdir", "test.json");
+            File.WriteAllText(testFile, "{}");
+
+            TestContext.BuiltDotNet.Exec("./subdir/test.json")
```

Comment:
Pull out `./subdir/test.json` into a separate variable and use it here and in the HaveStdErrContaining check.